Sources/SwiftRaft/Consensus/Raft.swift

```diff
@@ -225,14 +225,107 @@ extension Raft {
+            /// Current leader term id. Followers use them to validate correctness
+            public let termId: Term.ID
+
+            /// Leader id in the cluster
+            public let leaderId: NodeID
+
+            /// Index of log entry immediately preceding new ones
+            public let prevLogIndex: UInt64
+
+            /// Term id of prevLogIndex entry
+            public let prevLogTerm: UInt64
+
+            /// Leader’s commit index
+            public let leaderCommit: UInt64
+
+            /// Log entries to store (empty for heartbeat; may send more than one for efficiency)
+            public let entries: [LogElement<T>]
+
+            public init(termId: Term.ID,
+                        leaderId: NodeID,
+                        prevLogIndex: UInt64,
+                        prevLogTerm: UInt64,
+                        leaderCommit: UInt64,
+                        entries: [LogElement<T>]) {
+                self.termId = termId
+                self.leaderId = leaderId
+                self.prevLogIndex = prevLogIndex
+                self.prevLogTerm = prevLogTerm
+                self.leaderCommit = leaderCommit
+                self.entries = entries
+            }
+        }
+
+        public struct Response {
+            /// Current node term, for leader to update itself
+            public let termId: Term.ID
+
+            /// True if a follower accepted the message
+            public let success: Bool
+
+            public init(termId: Term.ID, success: Bool) {
+                self.termId = termId
+                self.success = success
+            }
+        }
+    }
+
+    public enum EntriesCommand: Equatable {
+        /// For non leader state reset an election timer
+        case resetElectionTimer
+    }
+
     /// Process Entry append
     ///
     /// 1. Reply false if term `<` currentTerm (§5.1)
     /// 2. Reply false if log doesn’t contain an entry at prevLogIndex whose term matches prevLogTerm (§5.3)
     /// 3. If an existing entry conflicts with a new one (same index but different terms), delete the existing entry and all that follow it (§5.3)
     /// 4. Append any new entries not already in the log
     /// 5. If leaderCommit > commitIndex, set commitIndex = min(leaderCommit, index of last new entry)
-    func onAppendEntries() async -> Void {
-        // TODO
+    /// Return a pair of side-effects and a response
+    func onAppendEntries<T: LogData>(_ request: AppendEntries.Request<T>) async -> ([EntriesCommand], AppendEntries.Response) {
+        var commands = Array<EntriesCommand>()
+        if request.termId > self.term.id {
+            // got a message with higher term, step down to a follower
+            do {
+                try term.tryToUpdateTerm(newTerm: request.termId, from: request.leaderId)
+                if !_tryMoveTo(nextState: .follower) {
+                    logger.debug("Got term greater than current and failed to move to the follower state")
```

Comment:
make sure to log the values, best as metadata

`logger.debug("appendEntries, incoming term >= currentTerm", metadata: ["incoming.term.id": ..., "term.id": ...])`

etc?

Reply:
Wil add